pkg/model/multiversion/mapper.go

```diff
@@ -0,0 +1,194 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package multiversion
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"gopkg.in/src-d/go-git.v4"
+
+	ackgenconfig "github.com/aws-controllers-k8s/code-generator/pkg/generate/config"
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+	"github.com/aws-controllers-k8s/code-generator/pkg/util"
+)
+
+var (
+	ErrAPIVersionNotFound   = errors.New("api version not found")
+	ErrAPIVersionDeprecated = errors.New("api version deprecated")
+)
+
+// ModelsMapper is a multi-version model mapper. It is containing the mapping
+// of each non-deprecated version with their correspending ackmodel.model
+// and APIInfos.
+type ModelsMapper struct {
+	gitRepo *git.Repository
+
+	hubVersion    string
+	spokeVersions []string
+
+	deprecatedVersions []string
+	removedVersions    []string
```

Comment:
I think we need a repository metadata configuration file. It should contain general metadata about the service - full name, short name, link to docs, etc. - but also a list of each version and their support state (deprecated, stable, under development). This would serve as the basis for the generated documentation as well.